File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string ans = "";
        int smallest = strs[0].size() ;
        for(string x : strs){
            if(x.size()<smallest){
                smallest = x.size();
            }
        }
        int i = 0 ;
        while (i < smallest){
            char c = strs[0][i];
            for(string x : strs){
                if(c != x[i]){
                    return ans ;
                }
            }
            ans = ans + c;
            i++;
        }
        return ans ;
    }
};

File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        word = strs[0]

        for x in range(len(word)) :
            for i in strs[1:] :
                if len(i) <= x or word[x] != i[x] :
                    return word[:x]

        return word 
